agent/agent.go

```diff
@@ -206,6 +206,11 @@ func (a *Agent) run(ctx context.Context) {
 		leaving       = a.leaving
 		subscriptions = map[string]context.CancelFunc{}
 	)
+	defer func() {
+		if session != nil {
```

Comment:
I don't think it's possible for `session` to be nil. In the `<-ctx.Done()` branch, `session.close()` is called without this check.

That call should probably be removed now that closing the session is handled in a `defer`. It's not harmful, but it's redundant.